File: frontend/src/types/RequirementSummaryDTO.ts
import type { RequirementState } from "./enum/RequirementState";

export interface RequirementSummaryDTO {
  id:number;
  entityIdentifier: string;

  name: string;
  description: string;
  state: RequirementState;
  requirementType: string;
}
export interface FunctionalRequirementSummaryDTO extends RequirementSummaryDTO {
  requirementType: "FR";
  functionalityId: number;
}
export type RequirementType = "FR"|"NFR";

File: backend/src/main/java/com/y4vra/irboardbackend/application/dtos/summaries/RequirementSummaryDTO.java
package com.y4vra.irboardbackend.application.dtos.summaries;

import java.io.Serializable;

public record RequirementSummaryDTO(
        Long id,
        String name,
        String description,
        String state,
        String requirementType  // "FR" or "NFR"
) implements RequirementSummary, Serializable {}
